src/sumset.c

```diff
@@ -168,6 +168,8 @@ void rs_signature_done(rs_signature_t *sig)
 {
     hashtable_free(sig->hashtable);
     rs_bzero(sig, sizeof(*sig));
+
+    free(sig->block_sigs);
```

Comment:
You need to do this before the sig is zero'ed... so put it after hashtable_free() and before rs_bzero().